pxtlib/tutorial.ts

```diff
@@ -305,6 +306,17 @@ ${code}
         return { header, hint, requiredBlocks };
     }
 
+    function categorizingValidationRules(listOfRules: pxt.Map<boolean>, title: string) {
+        const ruleNames = Object.keys(listOfRules);
+        for (let i = 0; i < ruleNames.length; i++) {
+            const setValidationRule: pxt.Map<string | number> = {};
+            setValidationRule["ruleName"] = ruleNames[i];
+            setValidationRule["enabled"] = listOfRules[ruleNames[i]] ? 'true' : 'false';
+            setValidationRule["tutorial"] = title;
```

Comment:
```suggestion
            const setValidationRule: pxt.Map<string> = {
                ruleName: ruleNames[i],
                enabled: listOfRules[ruleNames[i]] ? 'true' : 'false',
                tutorial: title,
            };
```

nit: usually a bit easier to use an an object literal vs explicitly indexing into it and setting properties, as it cuts out a bunch repetition / focuses attention more on the key itself / the values associated with them